packages/@react-spectrum/dialog/src/DialogTrigger.tsx

```diff
@@ -144,13 +144,48 @@ export {_DialogTrigger as DialogTrigger};
 
 function PopoverTrigger({state, targetRef, trigger, content, hideArrow, isKeyboardDismissDisabled, ...props}) {
   let triggerRef = useRef<HTMLElement>();
-
   let overlayRef = useRef<DOMRefValue<HTMLDivElement>>();
+  let isCloserPadding = false;
+
+  // For popover triggers close to the edge, decrease the container padding allowing popovers to
+  // move closer to their trigger.
+  if (triggerRef?.current) {
+    // The popover ref is not defined here, we don't know the arrow size to factor into the below.
+    let triggerRect = triggerRef.current.getBoundingClientRect();
+    let documentRect = document.body.getBoundingClientRect();
+    // 'bottom' is the default when no value is provided
+    let [axisPlacement] = props.placement?.split(' ') || ['bottom']; // Flip and RTL don't matter
+
+    // The following is the math that gets the desired beavhior in both scales, but isn't as logical
+    // as desired. For example, would love to not multiple the default padding by 2.
+    if (axisPlacement === 'start' || axisPlacement === 'end' || axisPlacement === 'right' || axisPlacement === 'left') {
+      // top edge start/end math
+      if ((DEFAULT_MODAL_PADDING * 2) - (triggerRect.height / 2) > triggerRect.top) {
+        isCloserPadding = true;
+      }
+      // bottom edge start/end math
+      // The documentRect doesn't work for this which is why we're using the window.innerHright.
```

Comment:
```suggestion
      // The documentRect doesn't work for this which is why we're using the window.innerHeight.
```

Reply:
why doesn't the document.body.getBoundingClientRect work? and why does the width work? I'm assuming this has to do with scrollable pages? we could have a horizontally scrolling page as well though, should we use innerWidth?

Do we actually want clientHeight? https://developer.mozilla.org/en-US/docs/Web/API/Window/innerHeight#usage_notes

Reply:
I remember that the triggerRef.current.getBoundingClientRect() and document.body.getBoundingClientRect() height and bottom values moved in relation to each other with a fixed difference instead of changing as the button is moved in and out of the viewable area.

Reply:
Two notes:

1 - clientHeight is the height of the viewable area. We are currently calculating where the button is relative to that edge. innerHeight is the total height of the page.

2 - The two bottom values are always different by 24px. See the images below. This 24px is 4px (36px height - 32px button height) of height in the container and 20px of bottom margin.

<img width="586" alt="image" src="https://user-images.githubusercontent.com/309540/196564476-25d341cd-d32b-4491-bc29-40816352d32e.png">
<img width="614" alt="image" src="https://user-images.githubusercontent.com/309540/196564750-ba8b762e-da49-42a7-a974-4e6121f47642.png">